news_r3k.cpp: Add support for the NWS-3410

briceonk opened the pull request:
This will probably need some work before merging, but opening it up for review early since it is touching pre-existing code. This refactors the news_r3k driver so that the LCD panel is only present on the laptop models (for now, just the 3260 but hopefully more like the 3150 bigNEWS will show up some day :)) and adds the ROM metadata for the 3410. It boots to the monitor ROM just fine, but NetBSD hangs during SCSI init :(

One thing I need to check with the owner of the machine (and will update on this PR once I hear back) is if the IDROM dump is available. I tested this with the 3260 IDROM patched to look like a 3410.

cuavas reviewed (commented):
I’m pretty tired at the moment, but I can’t see anything badly wrong with this at a glance.  @galibert and/or @pmackinlay do you want to take a look?

galibert commented:
That looks very clean to me.

pmackinlay commented:
Yes it's excellent and I'm happy to merge as soon as we have confirmed the name to credit.

briceonk commented:
The dump credit should go to `gerdavax` (cc @gerdavax :)). Working on the ID-ROM stuff, which should be the last thing needed for this PR. Will update soon.

rb6502 commented:
@briceonk What's the status on the ID-ROM?  We're getting towards the freeze on Saturday and it'd be great to have this in.

briceonk commented:
> @briceonk What's the status on the ID-ROM?  We're getting towards the freeze on Saturday and it'd be great to have this in.

Still waiting for the dump, unfortunately.

cuavas commented:
@briceonk this PR is already conflicting with master.  Can we get the code changes in and add the IDPROM later when the dump becomes available?

briceonk commented:
> @briceonk this PR is already conflicting with master.  Can we get the code changes in and add the IDPROM later when the dump becomes available?

I just so happened to get the IDPROM data this morning, so I will fix the merge conflicts and get the new metadata in this morning :)

briceonk commented:
> @briceonk this PR is already conflicting with master. Can we get the code changes in and add the IDPROM later when the dump becomes available?

OK, this should be ready! I fixed the merge conflicts and added the correct metadata. I'll send the bin file in a bit. Thanks to @gerdavax for getting that data!

![image](https://user-images.githubusercontent.com/32155223/221425155-e9675e2f-8c1d-4a1c-a726-7af63903bb02.png)

cuavas commented on src/mame/sony/news_r3k.cpp:
```diff
@@ -523,8 +608,18 @@ ROM_START(nws3260)
-COMP(1991, nws3260, 0,      0,      nws3260, nws3260, news_r3k_state, init_common, "Sony",  "NWS-3260", MACHINE_NO_SOUND)
+/*   YEAR  NAME     PARENT  COMPAT  MACHINE  INPUT    CLASS          INIT         COMPANY  FULLNAME    FLAGS */
+COMP(1991, nws3260, 0,      0,      nws3260, nws3260, nws3260_state, init_common, "Sony",  "NWS-3260", MACHINE_NO_SOUND)
+COMP(1991, nws3410, 0,      0,      nws3410, nws3410, nws3410_state, init_common, "Sony",  "NWS-3410", MACHINE_NO_SOUND)
```
I’ve already merged this before it rots again, but should this system have an unimplemented graphics flag due to lack of framebuffer emulation?  I can add the flag myself.

briceonk replied:
Oh, yeah, it should. Sorry, I must have forgotten to add that myself :(
